impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -192,19 +256,68 @@ DownsamplePassArgs CalculateDownsamplePassArgs(
   //
   //   !input_snapshot->GetCoverage()->Expand(-local_padding)
   //     .Contains(coverage_hint.value()))
-  Vector2 downsampled_size = source_rect_padded.GetSize() * downsample_scalar;
-  ISize subpass_size =
-      ISize(round(downsampled_size.x), round(downsampled_size.y));
-  Vector2 effective_scalar =
-      Vector2(subpass_size) / source_rect_padded.GetSize();
-
-  Quad uvs = GaussianBlurFilterContents::CalculateUVs(
-      input, snapshot_entity, source_rect_padded, input_snapshot_size);
-  return {
-      .subpass_size = subpass_size,
-      .uvs = uvs,
-      .effective_scalar = effective_scalar,
-  };
+
+  std::optional<Rect> snapshot_coverage = input_snapshot.GetCoverage();
+  if (input_snapshot.transform.IsIdentity() &&
+      source_expanded_coverage_hint.has_value() &&
+      snapshot_coverage.has_value() &&
+      snapshot_coverage->Contains(source_expanded_coverage_hint.value())) {
+    // If the snapshot's transform is the identity transform and we have
+    // coverage hint that fits inside of the snapshots coverage that means the
+    // coverage hint was ignored so we will trim out the area we are interested
+    // in the down-sample pass. This usually means we have a backdrop image
+    // filter.
+    //
+    // The region we cut out will be aligned with the down-sample divisor to
+    // avoid pixel alignment problems that create shimmering.
+    int32_t divisor = std::round(1.0f / desired_scalar);
+    Rect aligned_coverage_hint = Rect::MakeLTRB(
+        FloorToDivisible(source_expanded_coverage_hint->GetLeft(), divisor),
+        FloorToDivisible(source_expanded_coverage_hint->GetTop(), divisor),
+        source_expanded_coverage_hint->GetRight(),
+        source_expanded_coverage_hint->GetBottom());
+    aligned_coverage_hint = Rect::MakeXYWH(
+        aligned_coverage_hint.GetX(), aligned_coverage_hint.GetY(),
+        CeilToDivisible(aligned_coverage_hint.GetWidth(), divisor),
+        CeilToDivisible(aligned_coverage_hint.GetHeight(), divisor));
+    ISize source_size = ISize(aligned_coverage_hint.GetSize().width,
+                              aligned_coverage_hint.GetSize().height);
+    Vector2 downsampled_size = source_size * downsample_scalar;
+    Scalar int_part;
```

Comment:
You can also add `[[maybe_unused]]`

Reply:
I think I did that in the past and the compiler for fuchsia or windows didn't care =T